renku/cli/_graph.py

```diff
@@ -67,62 +67,55 @@ def __attrs_post_init__(self):
     @_nodes.default
     def default_nodes(self):
         """Build node index."""
-        _nodes = OrderedDict()
-        _need_update = self._need_update = {}
-        _latest_commits = self._latest_commits
+        nodes = []
+        for commit in reversed(self._sorted_commits):
+            try:
+                activity = self.commits[commit]
+                nodes.extend(reversed(list(activity.nodes)))
```

Comment:
don't need `list`

Reply:
it does need the `list` since `activity.nodes` might be a generator.

Reply:
I know, but `extend` will take care of it, no?

Reply:
but not `reversed`

Reply:
`reversed` can take a generator:

```
In [1]: reversed(range(10))
Out[1]: <range_iterator at 0x10acd79c0>

In [2]: x = []

In [3]: x.extend(reversed(range(10)))

In [4]: x
Out[4]: [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
```

Reply:
I stand corrected, `range` is a special case ;)